trieste/models/gpflux/models.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import Any, Dict
+
+import tensorflow as tf
+from gpflow.inducing_variables import InducingPoints
+from gpflux.layers import GPLayer, LatentVariableLayer
+from gpflux.models import DeepGP
+
+from ...data import Dataset
+from ...types import TensorType
+from ..interfaces import TrainableProbabilisticModel
+from .interface import GPfluxPredictor
+from .utils import sample_dgp
+
+
+class DeepGaussianProcess(GPfluxPredictor, TrainableProbabilisticModel):
+    """
+    A :class:`TrainableProbabilisticModel` wrapper for a GPflux :class:`~gpflux.models.DeepGP` with
+    :class:`GPLayer` or :class:`LatentVariableLayer`: this class does not support e.g. keras layers.
+    Note: the user should remember to set `tf.keras.backend.set_floatx()` with the desired value
+    (consistent with GPflow) so that dtype errors do not occur.
+    """
+
+    def __init__(
+        self,
+        model: DeepGP,
+        optimizer: tf.optimizers.Optimizer | None = None,
+        fit_args: Dict[str, Any] | None = None,
+    ):
+        """
+        :param model: The underlying GPflux deep Gaussian process model.
+        :param optimizer: The optimizer with which to train the model. Defaults to
+            :class:`~trieste.models.optimizer.TFOptimizer` with :class:`~tf.optimizers.Adam`. Only
+            the optimizer itself is used; other args relevant for fitting should be passed as part
+            of `fit_args`.
+        :param fit_args: A dictionary of arguments to be used in the Keras `fit` method. Default to
```

Comment:
This needs a link to Keras docs that lists possible args.

```suggestion
        :param fit_args: A dictionary of arguments to be used in the Keras `fit` method. Defaults to
```

Reply:
Done